recommend/imageHandler.go

```diff
@@ -477,37 +479,91 @@ func getImageDetails(img ImageInfo) error {
 	// step 3: getImageInfo
 	img.getImageInfo()
 
+	if len(img.RepoTags) == 0 {
+		img.RepoTags = append(img.RepoTags, img.Name)
+	}
 	// step 4: get policy from image info
 	img.getPolicyFromImageInfo()
 
 	return nil
 }
 
-func imageHandler(namespace, deployment string, labels LabelMap, imageName string, config string) error {
-	dockerConfigPath = config
-	log.WithFields(log.Fields{
-		"image": imageName,
-	}).Info("pulling image")
+func imageHandler(namespace, deployment string, labels LabelMap, imageName string, c *k8s.Client) error {
+	dockerConfigPath = options.Config
 	img := ImageInfo{
 		Name:       imageName,
 		Namespace:  namespace,
 		Deployment: deployment,
 		Labels:     labels,
 	}
+	policiesToBeRecommended := options.Policy
+
+	if len(policiesToBeRecommended) == 1 {
+		policyToBeRecommended := policiesToBeRecommended[0]
+		if policyToBeRecommended == KyvernoPolicy {
+			if len(img.RepoTags) == 0 {
+				img.RepoTags = append(img.RepoTags, img.Name)
+			}
+			if err := ReportStart(&img); err != nil {
+				log.WithError(err).Error("report start failed")
+				return err
+			}
+			err := recommendAdmissionControllerPolicies(img, c)
+			if err != nil {
+				log.WithError(err).Error("failed to recommend admission controller policies.")
+				return err
+			}
+		} else if policyToBeRecommended == KubeArmorPolicy {
+			err := recommendKubeArmorPolicies(imageName, img)
+			if err != nil {
+				log.WithError(err).Error("failed to recommend kubearmor policies.")
+				return err
+			}
+		} else {
+			return fmt.Errorf("policy of kind %s cannot be generated", policyToBeRecommended)
+		}
+	} else if len(policiesToBeRecommended) == 2 {
+		if reflect.DeepEqual(policiesToBeRecommended, DefaultPoliciesToBeRecommended) ||
+			reflect.DeepEqual(policiesToBeRecommended, []string{KubeArmorPolicy, KyvernoPolicy}) {
+			errKubeArmorPol := recommendKubeArmorPolicies(imageName, img)
+			if errKubeArmorPol != nil {
+				log.WithError(errKubeArmorPol).Error("failed to recommend kubearmor policies.")
+			}
+			if len(img.RepoTags) == 0 {
+				img.RepoTags = append(img.RepoTags, img.Name)
+			}
+			errAdmissionControllerPol := recommendAdmissionControllerPolicies(img, c)
+			if errAdmissionControllerPol != nil {
+				log.WithError(errAdmissionControllerPol).Error("Failed to recommend admission controller policies.")
+			}
+			if errKubeArmorPol != nil || errAdmissionControllerPol != nil {
+				return fmt.Errorf("failed to recommend policies for image %s", imageName)
+			}
+		}
+	} else {
+		return fmt.Errorf("policies of kind %v can't be generated", policiesToBeRecommended)
+	}
```

Comment:
This whole logic needs heavy refactoring.

Reply:
Refactored, done.